plugins/woocommerce/src/Blocks/Domain/Services/CheckoutFieldsFrontend.php

```diff
@@ -185,23 +185,33 @@ public function edit_account_form_fields() {
 	 *
 	 * @param integer $user_id User ID.
 	 */
-	public function save_account_form_fields( $user_id ) {
-		// phpcs:disable WordPress.Security.NonceVerification.Missing
-		$customer          = new WC_Customer( $user_id );
+	public function save_account_form_fields( int $user_id ) {
+
+		try {
+			$customer = new WC_Customer( $user_id );
+		} catch ( \Exception $e ) {
+			wc_add_notice(
+				__( 'Unable to save customer additional fields: ', 'woocommerce' ) . $e->getMessage(),
+				'notice'
+			);
+			return;
+		}
+
 		$additional_fields = $this->checkout_fields_controller->get_fields_for_location( 'contact' );
-		$field_values      = array();
 
-		foreach ( array_keys( $additional_fields ) as $key ) {
-			$post_key = CheckoutFields::get_group_key( 'other' ) . $key;
-			if ( ! isset( $_POST[ $post_key ] ) ) {
-				continue;
-			}
+		$field_values = array();
 
-			$field_value = $this->checkout_fields_controller->sanitize_field( $key, wc_clean( wp_unslash( $_POST[ $post_key ] ) ) );
+		foreach ( $additional_fields as $key => $field_data ) {
+			// We can't skip, field might be required.
+			$field_value = wc_clean( wp_unslash( $_POST[ $key ] ?? '' ) ); // phpcs:ignore WordPress.Security.NonceVerification.Missing
+
+			$field_value = $this->checkout_fields_controller->sanitize_field( $key, $field_value );
 			$validation  = $this->checkout_fields_controller->validate_field( $key, $field_value );
 
 			if ( is_wp_error( $validation ) && $validation->has_errors() ) {
-				wc_add_notice( $validation->get_error_message(), 'error' );
+				foreach ( $validation->get_error_messages() as $error_message ) {
+					wc_add_notice( $error_message, 'error' );
+				}
```

Comment:
I know messages come condensed,  but since these are using callbacks that can be custom-defined, we should account for `WP_Error` possibly being used to the full extent, or else risk suppressing error messages. Right?